d2/src/test/java/com/linkedin/d2/xds/TestXdsClientImpl.java

```diff
@@ -700,51 +699,47 @@ public void testResourceSubscriberAddWatcher()
     verify(_wildcardWatcher, times(10)).onChanged(eq(CLUSTER_RESOURCE_NAME), eq(CLUSTER_NAME_DATA_UPDATE));
   }
 
-  @DataProvider(name = "provideUseGlobCollection")
-  public Object[][] provideUseGlobCollection()
+  @DataProvider(name = "provideUseGlobCollectionAndIRV")
+  public Object[][] provideUseGlobCollectionAndIRV()
   {
     // {
     //   useGlobCollection --- whether to use glob collection
     // }
     return new Object[][]{
-        {true},
-        {false}
+        {true, true},
+        {false, false}
     };
   }
 
-  @Test(dataProvider = "provideUseGlobCollection", timeOut = 2000)
+  @Test(dataProvider = "provideUseGlobCollectionAndIRV", timeOut = 2000)
   // Retry task should re-subscribe the resources registered in each subscriber type.
-  public void testRetry(boolean useGlobCollection) throws ExecutionException, InterruptedException
+  public void testRetry(boolean useGlobCollection, boolean useIRV) throws ExecutionException, InterruptedException
   {
-    XdsClientImplFixture fixture = new XdsClientImplFixture(useGlobCollection);
+    XdsClientImplFixture fixture = new XdsClientImplFixture(useGlobCollection, useIRV);
     fixture.watchAllResourceAndWatcherTypes();
     fixture._xdsClientImpl.testRetryTask(fixture._adsStream);
     fixture._xdsClientImpl._retryRpcStreamFuture.get();
 
     // get all the resource types and names sent in the discovery requests and verify them
     List<ResourceType> types = fixture._resourceTypesArgumentCaptor.getAllValues();
-    List<String> nameLists = fixture._resourceNamesArgumentCaptor.getAllValues().stream()
-        .map(names ->
-        {
-          if (names.size() != 1)
-          {
-            Assert.fail("Resource names should be a singleton list");
-          }
-          return names.iterator().next();
-        }).collect(Collectors.toList());
-
-    Set<Pair<ResourceType, String>> args = IntStream.range(0, types.size())
-        .mapToObj(i -> Pair.of(types.get(i), nameLists.get(i)))
-        .collect(Collectors.toSet());
-
-    Assert.assertEquals(args, ImmutableSet.of(
-        Pair.of(NODE, "*"),
-        Pair.of(NODE, SERVICE_RESOURCE_NAME),
-        Pair.of(useGlobCollection ? D2_URI : D2_URI_MAP, "*"),
-        Pair.of(useGlobCollection ? D2_URI : D2_URI_MAP, useGlobCollection ? CLUSTER_GLOB_COLLECTION : CLUSTER_RESOURCE_NAME),
-        Pair.of(D2_URI, URI_URN1),
-        Pair.of(D2_CLUSTER_OR_SERVICE_NAME, "*")
-    ));
+    List<Collection<String>> nameLists = fixture._resourceNamesArgumentCaptor.getAllValues();
+
+    Map<ResourceType, Set<String>> resourceNames = new HashMap<>();
+    for (int i = 0; i < types.size(); i++) {
+      resourceNames.computeIfAbsent(types.get(i), k -> new HashSet<>()).addAll(nameLists.get(i));
+    }
+
+    Assert.assertEquals(resourceNames.get(NODE), ImmutableSet.of(SERVICE_RESOURCE_NAME, "*"));
```

Comment:
as IRV is added now, you need to verify the actual IRVs being sent in the request, otherwise it's not testing IRV at all.